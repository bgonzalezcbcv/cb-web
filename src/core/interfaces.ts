import { NavigateOptions } from "react-router-dom";

export interface VisualComponent {
	width?: string;
	height?: string;
}

export interface Teacher {
	ci: string;
	firstName: string;
	lastName: string;
	subjects: string[];
}

export enum UserRoleName {
	Administrador = "Administrador",
	Administrativo = "Administrativo",
	Adscripto = "Adscripto",
	Director = "Director",
	Docente = "Docente",
	Recepcion = "Recepci√≥n",
}

export enum UserRoleColor {
	Administrador = "red",
	Administrativo = "blue",
	Adscripto = "cyan",
	Director = "green",
	Docente = "yellow",
	Recepcion = "black",
}

export enum UserRole {
	Administrador,
	Administrativo,
	Adscripto,
	Director,
	Docente,
	Recepcion,
}

export interface User {
	email: string;
	token: string;
	displayName: string;
	role: UserRole;
}

export interface SidebarItem {
	title: string;
	navigationRoute: string;
	navigationParams?: NavigateOptions;
}

export interface SidebarSection {
	sectionTitle: string;
	items: SidebarItem[];
}

export interface Question {
	id: string;
	question: string;
	answer: string;
}

export interface QuestionCategories {
	category: string;
	questions: Question[];
}

export interface Student {
	question_categories: QuestionCategories[];
}
