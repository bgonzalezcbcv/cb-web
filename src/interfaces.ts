export interface VisualComponent {
    width?: string,
    height?: string
}