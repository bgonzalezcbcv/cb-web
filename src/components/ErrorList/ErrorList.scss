.error-list {
	width: max-content;

	ul {
		list-style-type: none;
		margin-top: 0;
	}
}
