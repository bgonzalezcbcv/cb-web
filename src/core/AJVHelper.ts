import _ from "lodash";
import Ajv, { ErrorObject } from "ajv";
import ajvErrors from "ajv-errors";
import ajvFormat from "ajv-formats";

const newAjv = new Ajv({ allErrors: true, verbose: true, strictRequired: true });

export const ajv = ajvErrors(ajvFormat(newAjv));

export function getAjvErrors(ajvValidator = ajv): ErrorObject[] | null | undefined {
	console.log(ajv.errors);
	return ajvValidator.errors;
}

// Used to fill the ErrorList component
export function getParsedErrors(ajvValidator = ajv): Record<string, unknown> {
	if (ajvValidator.errors?.length === 0) return {};
	let requiredId = 0;

	const parsedErrors = ajvValidator.errors?.map((error) => {
		if (error.keyword === "required") {
			requiredId++;
			const missingPropertyName = error.params.missingProperty;
			const title = error.parentSchema?.properties[missingPropertyName].title;

			return {
				id:
					error.instancePath
						.replace("/", "")
						.replaceAll(/\/(\d*)\//gm, "[$1]/")
						.replaceAll("/", ".")
						.replaceAll(/\.(\d*)\./gm, "[$1].") +
					(error.instancePath === "" ? `` : ".") +
					`required${requiredId}`,
				title,
				message: "Campo requerido.",
			};
		}

		return {
			id: error.instancePath
				.replace("/", "")
				.replaceAll(/\/(\d*)\//gm, "[$1]/")
				.replaceAll("/", "."),
			title: error.parentSchema?.title,
			message: error.message,
		};
	});

	console.log(parsedErrors);

	let errorObject = {};

	parsedErrors?.forEach((error) => {
		const { id, title, message } = error;

		errorObject = _.set(errorObject, id, {
			title,
			message,
		});
	});

	return errorObject;
}
