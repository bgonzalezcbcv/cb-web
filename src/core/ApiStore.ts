import axios, { Axios } from "axios";

import { Student } from "./Models";
import { User, UserRole } from "./interfaces";

import { DataStore } from "./DataStore";

const dataStore = DataStore.getInstance();

const baseConfig = {
	headers: {
		Authorization: `Bearer ${dataStore.loggedUser?.token}`,
		"Content-Type": "application/json",
	},
	baseURL: process.env["REACT_APP_API_URL"],
};

// todo: use this instance in order to do a middleware.
//eslint-disable-next-line
const axiosInstance = new Axios(baseConfig);

// todo: create middleware in order to log out on 403.

interface SignInResponseData {
	id: number;
	email: string;
	ci: string;
	name: string;
	surname: string;
	birthdate: string;
	address: string;
	role?: UserRole;
}
//todo: make the logout.
export async function login(email: string, password: string): Promise<{ success: boolean; data?: User; err: string }> {
	try {
		const errObject = {
			success: false,
			err: "Error durante el inicio de sesi칩n, volver a intentarlo.",
		};

		const config = {
			...baseConfig,
			method: "post",
			url: "/api/sign_in/",
			headers: {
				"Content-Type": "application/json",
			},
			data: JSON.stringify({
				email,
				password,
			}),
		};

		const response = await axios(config);

		if (!(response.data && response.headers)) return errObject;

		const { name, surname, role }: SignInResponseData = response.data;

		const [bearer, token] = (response.headers["authorization"] ?? "").split(" ");

		if (bearer !== "Bearer") return errObject;

		return { success: true, data: { email, token, name, surname, role: role ?? UserRole.Administrador }, err: "" };
		// eslint-disable-next-line
	} catch (error: any) {
		let err = "";
		if (!error?.response) {
			err = "El servidor no responde";
		} else if (error.response?.status === 400) {
			err = "Usuario o contrase침a inv치lidos";
		} else if (error.response?.status === 401) {
			err = "No autorizado";
		} else {
			err = "Fall칩 el inicio";
		}
		return { success: false, err };
	}
}

export async function createStudent(studentToCreate: Student): Promise<boolean> {
	try {
		const config = {
			...baseConfig,
			method: "post",
			url: "/api/students/",
			data: JSON.stringify({
				student: studentToCreate,
			}),
		};

		const response = await axios(config);

		return response.status === 201;
	} catch (e) {
		return false;
	}
}

// // eslint-disable-next-line
// export async function login(user: string, pass: string): Promise<{ success: boolean; data?: User; err: string }> {
// 	return { success: true, data: { email: "pepe@gmail.com", name: "Pepe Bolsilludo", role: UserRole.Administrador, token: "Bolsilludo" }, err: "" };
// }
