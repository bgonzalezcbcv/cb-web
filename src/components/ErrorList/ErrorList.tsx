import _ from "lodash";
import React from "react";

import { getTitleFromSchema } from "../../core/CoreHelper";
import { JsonSchema7 } from "@jsonforms/core";
import { Box, List, ListItem, Typography } from "@mui/material";

import "./ErrorList.scss";

interface ErrorRow {
	title: string;
	message: string;
}

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
type ErrorsEntry = Record<string, ErrorRow | ErrorsEntry>;

interface ErrorListProps {
	name?: string;
	path: string;
	value: ErrorsEntry;
	schema: JsonSchema7;
}

function ErrorList(props: ErrorListProps): JSX.Element {
	const { name, path, value, schema } = props;

	function printRow(name: string, message: string): React.ReactElement {
		return (
			<ListItem key={`error-row-${name}`}>
				<Box display="flex" flexDirection="row" gap="5px">
					<Typography fontWeight="bold" color="darkred">{`${name}: `}</Typography>
					<Typography>{`${message}`}</Typography>
				</Box>
			</ListItem>
		);
	}

	function printObject(name: string, obj: Record<string, unknown>): React.ReactElement {
		return (
			<ListItem className="error-list">
				<Box>
					{name ? (
						<Typography fontWeight="bold" color="darkred" fontSize="medium">
							{name}
						</Typography>
					) : null}

					<List>
						{Object.entries(obj).map((entry, index) => (
							<ErrorList
								key={`error-row-obj-${index}`} //
								path={path !== "" ? path + "." + entry[0] : entry[0]}
								value={entry[1]}
								schema={schema}
							/>
						))}
					</List>
				</Box>
			</ListItem>
		);
	}

	function printArray(name: string, arr: Record<string, unknown>[]): React.ReactElement {
		return (
			<ListItem>
				<Box>
					<Typography fontWeight="bold" color="darkred" fontSize="medium">
						{name}
					</Typography>

					<List>
						{arr.map((entry, index) => (
							<ErrorList
								key={`error-row-obj-${index}`} //
								path={path + `[${index}]`}
								name={`${index + 1} Â°`}
								value={entry}
								schema={schema}
							/>
						))}
					</List>
				</Box>
			</ListItem>
		);
	}

	if (value.message) return printRow(name ?? value.title, value.message);

	if (_.isArray(value)) return printArray(name ?? getTitleFromSchema(path, schema) ?? "", value);

	if (_.isPlainObject(value)) return printObject(name ?? getTitleFromSchema(path, schema) ?? "", value as Record<string, unknown>);

	return <></>;
}

export default ErrorList;
