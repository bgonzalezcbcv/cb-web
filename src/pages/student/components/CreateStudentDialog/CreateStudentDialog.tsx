import React, { useCallback, useState } from "react";

import { Student } from "../../../../core/Models";
import * as API from "../../../../core/ApiStore";
import { ajv as studentAjv, getAjvErrors } from "../../../../core/AJVHelper";
import { getParsedErrors } from "../../../../core/AJVHelper";

import { Button, Box, Dialog, DialogActions, DialogContent, DialogTitle, Divider, Typography, Alert } from "@mui/material";
import { LoadingButton } from "@mui/lab";
import ErrorList from "../../../../components/ErrorList/ErrorList";

import studentSchema from "../../schema.json";

interface CreateStudentDialogProps {
	student: Student;
}

function CreateStudentDialog(props: CreateStudentDialogProps): React.ReactElement {
	const { student } = props;

	const [isOpen, setIsOpen] = React.useState(false);
	const [studentCreationState, setStudentCreationState] = React.useState<"idle" | "inProcess" | "success" | "fail">("idle");
	const [errors, setErrors] = useState({});

	const handleStudentCreation = useCallback(async (): Promise<void> => {
		setStudentCreationState("inProcess");

		const successfulCreation = await API.createStudent(student);

		if (successfulCreation) {
			setStudentCreationState("success");
		} else {
			setStudentCreationState("fail");
			setIsOpen(true);
		}
	}, [student]);

	const dismiss = (): void => {
		setIsOpen(false);
		setStudentCreationState("idle");
	};

	const onCreateClickHandler = (): void => {
		studentAjv.validate(studentSchema, student);

		const errors = getAjvErrors(studentAjv);

		if (errors && errors.length > 0) {
			setIsOpen(true);
			setErrors(getParsedErrors(studentAjv));
		} else {
			handleStudentCreation();
		}
	};

	return (
		<>
			<Box>
				<Divider sx={{ marginBottom: "10px" }}></Divider>

				<Box display="flex" justifyContent="flex-end" alignContent="flex-end" alignSelf="flex-end" onClick={onCreateClickHandler}>
					<Button variant="outlined">Crear Alumno</Button>
				</Box>
			</Box>

			{isOpen ? (
				<Dialog open={isOpen} onClose={dismiss}>
					<DialogTitle>
						<Typography component={"span"} variant="h5" fontWeight="bold">
							Hay errores en los campos del alumno...
						</Typography>
					</DialogTitle>

					<DialogContent>
						{studentCreationState === "fail" ? (
							<Alert severity="error">No se pudo crear el alumno. Inténtelo de nuevo o corrija los errores.</Alert>
						) : (
							<Typography component={"span"}>¿Está seguro de querer crear este alumno?</Typography>
						)}

						<Box height="400px" overflow="auto" paddingTop="12px">
							<ErrorList name="Errores" path="" value={errors} schema={studentSchema} />
						</Box>
					</DialogContent>

					<DialogActions sx={{ display: "flex", justifyContent: "space-around" }}>
						<LoadingButton variant="outlined" onClick={dismiss} loading={studentCreationState === "inProcess"}>
							Cancelar
						</LoadingButton>

						<LoadingButton variant="outlined" onClick={handleStudentCreation} loading={studentCreationState === "inProcess"}>
							{studentCreationState === "fail" ? "Reintentar" : "Aceptar"}
						</LoadingButton>
					</DialogActions>
				</Dialog>
			) : null}

			{studentCreationState === "success" ? (
				<Dialog open={studentCreationState === "success"} onClose={dismiss}>
					<DialogTitle>¡Estudiante creado correctamente!</DialogTitle>

					<DialogActions>
						<Button variant="outlined" onClick={dismiss}>
							Aceptar
						</Button>
					</DialogActions>
				</Dialog>
			) : null}
		</>
	);
}

export default CreateStudentDialog;
