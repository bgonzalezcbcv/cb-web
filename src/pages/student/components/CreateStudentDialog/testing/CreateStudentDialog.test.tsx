import React from "react";
import { act, render } from "@testing-library/react";

import { Student } from "../../../../../core/Models";
import CreateStudentDialog from "../CreateStudentDialog";
import * as API from "../../../../../core/ApiStore";
import * as AJVHelper from "../../../../../core/AJVHelper";
import * as ErrorList from "../../../../../components/ErrorList/ErrorList";
import { ErrorObject } from "ajv";
import userEvent from "@testing-library/user-event";
import { StudentPageMode } from "../../../../../core/interfaces";

describe("CreateStudentDialog", () => {
	beforeEach(() => {
		jest.spyOn(ErrorList, "default").mockReturnValue(<div>Error List</div>);
		jest.spyOn(API, "createStudent").mockResolvedValue(true);
	});

	it("should render a button", () => {
		const wrapper = render(<CreateStudentDialog student={{} as Student} />);

		expect(wrapper).toMatchSnapshot();
	});

	it("should render a dialog when create student button is clicked", async () => {
		jest.spyOn(AJVHelper, "getAjvErrors").mockReturnValue([{} as ErrorObject]);
		jest.spyOn(AJVHelper, "getParsedErrors").mockReturnValue({});

		const wrapper = render(<CreateStudentDialog student={{} as Student} />);

		const createStudentButton = await wrapper.findByText("Crear Alumno");

		act(() => {
			userEvent.click(createStudentButton);
		});

		expect(wrapper).toMatchSnapshot();
	});

	it("should not render a dialog when create student button is clicked and no errors are found", async () => {
		jest.spyOn(AJVHelper, "getAjvErrors").mockReturnValue(null);
		jest.spyOn(AJVHelper, "getParsedErrors").mockReturnValue({});

		const wrapper = render(<CreateStudentDialog student={{} as Student} />);

		const createStudentButton = await wrapper.findByText("Crear Alumno");

		act(() => {
			userEvent.click(createStudentButton);
		});

		const successTitle = await wrapper.findByText("¡Estudiante creado correctamente!");

		expect(successTitle).toBeVisible();
	});

	it("should not render a dialog with errors when create student button is clicked and no errors are found but fails creation", async () => {
		jest.spyOn(AJVHelper, "getAjvErrors").mockReturnValue(null);
		jest.spyOn(AJVHelper, "getParsedErrors").mockReturnValue({});
		jest.spyOn(API, "createStudent").mockResolvedValue(false);

		const wrapper = render(<CreateStudentDialog student={{} as Student} />);

		const createStudentButton = await wrapper.findByText("Crear Alumno");

		act(() => {
			userEvent.click(createStudentButton);
		});

		const failureAlert = await wrapper.findByText("No se pudo crear el alumno. Inténtelo de nuevo o corrija los errores.");

		expect(failureAlert).toBeVisible();
	});

	it("should not render a dialog when cancel button is clicked", async () => {
		jest.spyOn(AJVHelper, "getAjvErrors").mockReturnValue([{} as ErrorObject]);
		jest.spyOn(AJVHelper, "getParsedErrors").mockReturnValue({});

		const wrapper = render(<CreateStudentDialog student={{} as Student} />);

		const createStudentButton = await wrapper.findByText("Crear Alumno");

		act(() => {
			userEvent.click(createStudentButton);
		});

		const cancelButton = await wrapper.findByText("Cancelar");

		act(() => {
			userEvent.click(cancelButton);
		});

		expect(cancelButton).not.toBeVisible();

		expect(wrapper).toMatchSnapshot();
	});

	it("should render a success dialog on creation success", async () => {
		jest.spyOn(AJVHelper, "getAjvErrors").mockReturnValue([{} as ErrorObject]);
		jest.spyOn(AJVHelper, "getParsedErrors").mockReturnValue({});
		jest.spyOn(API, "createStudent").mockResolvedValue(true);

		const wrapper = render(<CreateStudentDialog student={{} as Student} />);

		const createStudentButton = await wrapper.findByText("Crear Alumno");

		act(() => {
			userEvent.click(createStudentButton);
		});

		const acceptButton = await wrapper.findByText("Aceptar");

		act(() => {
			userEvent.click(acceptButton);
		});

		const successTitle = await wrapper.findByText("¡Estudiante creado correctamente!");

		expect(successTitle).toBeVisible();
	});

	it("should render a an alert on creation failure", async () => {
		jest.spyOn(AJVHelper, "getAjvErrors").mockReturnValue([{} as ErrorObject]);
		jest.spyOn(AJVHelper, "getParsedErrors").mockReturnValue({});
		jest.spyOn(API, "createStudent").mockResolvedValue(false);

		const wrapper = render(<CreateStudentDialog student={{} as Student} />);

		const createStudentButton = await wrapper.findByText("Crear Alumno");

		act(() => {
			userEvent.click(createStudentButton);
		});

		const acceptButton = await wrapper.findByText("Aceptar");

		act(() => {
			userEvent.click(acceptButton);
		});

		const failureAlert = await wrapper.findByText("No se pudo crear el alumno. Inténtelo de nuevo o corrija los errores.");
		const retryButton = await wrapper.findByText("Reintentar");

		expect(failureAlert).toBeVisible();
		expect(retryButton).toBeVisible();
	});
});
