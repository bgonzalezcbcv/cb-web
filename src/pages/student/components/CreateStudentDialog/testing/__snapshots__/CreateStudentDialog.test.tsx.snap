// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateStudentDialog should not render a dialog when cancel button is clicked 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-o9b5bv-MuiDivider-root"
        />
        <div
          class="MuiBox-root css-1doqpq5"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-wnpvqa-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Crear Alumno
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-o9b5bv-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-1doqpq5"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-wnpvqa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Crear Alumno
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CreateStudentDialog should render a button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-o9b5bv-MuiDivider-root"
        />
        <div
          class="MuiBox-root css-1doqpq5"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-wnpvqa-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Crear Alumno
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
    >
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-o9b5bv-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-1doqpq5"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-wnpvqa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Crear Alumno
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`CreateStudentDialog should render a dialog when create student button is clicked 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      <div
        class="MuiBox-root css-0"
      >
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-o9b5bv-MuiDivider-root"
        />
        <div
          class="MuiBox-root css-1doqpq5"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-wnpvqa-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Crear Alumno
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r0:"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1t1j96h-MuiPaper-root-MuiDialog-paper"
          role="dialog"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
            id=":r0:"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 css-bezmlo-MuiTypography-root"
            >
              Hay errores en los campos del alumno...
            </span>
          </h2>
          <div
            class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              ¿Está seguro de querer crear este alumno?
            </span>
            <div
              class="MuiBox-root css-10idku2"
            >
              <div>
                Error List
              </div>
            </div>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing css-1chrrgc-MuiDialogActions-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-yvzrdt-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
              tabindex="0"
              type="button"
            >
              Cancelar
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-bi5g9h-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
              tabindex="0"
              type="button"
            >
              Aceptar
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      class="MuiBox-root css-0"
    >
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-o9b5bv-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-1doqpq5"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-wnpvqa-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Crear Alumno
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
